src/strat_engine/blockdev.rs

```diff
@@ -55,87 +55,109 @@ impl BlockDev {
                mda_size: Sectors,
                force: bool)
                -> EngineResult<BlockDev> {
-        let metadata = try!(fs::metadata(path));
+        Ok(try!(Self::batch_new(stratis_id, &[path], mda_size, force)).pop().unwrap())
+    }
 
-        if !metadata.file_type().is_block_device() {
-            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} is not a block device",
-                                                              path.display()))));
-        }
+    /// Initialize multiple blockdevs at once. This allows all of them
+    /// to be checked for usability before writing to any of them.
+    pub fn batch_new(stratis_id: &str,
+                     paths: &[&Path],
+                     mda_size: Sectors,
+                     force: bool)
+                     -> EngineResult<Vec<BlockDev>> {
 
         if *mda_size % 2 != 0 {
             return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} mda size is not an even \
+                                                      format!("mda size {} is not an even \
                                                                number",
-                                                              path.display()))));
+                                                              *mda_size))));
         }
 
         if mda_size < MIN_MDA_SIZE {
             return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} mda size is less than \
+                                                      format!("mda size {} is less than \
                                                                minimum ({})",
-                                                              path.display(),
+                                                              *mda_size,
                                                               *MIN_MDA_SIZE))));
         }
 
-        let dev = try!(Device::from_str(&path.to_string_lossy()));
+        // Check stuff for all candidates before writing anything
+        let mut good_paths = Vec::new();
+        for path in paths {
+            let metadata = try!(fs::metadata(path));
```

Comment:
It would be nice if the chunk from line 87 to line 126 were a single function that returned an EngineResult. Then use ```find()``` to find the first path that is an error and return that error if so, otherwise proceed. This will make everything easier when we have to abstract dealing with paths that are duplicates or not paths at all and all those other issues. Please see if you can abstract the checking into a single function.